Assets/MRTK/Core/Utilities/Editor/XRSettingsUtilities.cs

```diff
@@ -73,24 +185,43 @@ public static bool IsLegacyXRActive
                     {
                         if (displayDescriptor.disablesLegacyVr)
                         {
-                            isLegacyXRActive = false;
+                            isXRSDKSuppressingLegacyXR = true;
                             break;
                         }
                     }
                 }
 
-                return isLegacyXRActive.HasValue && isLegacyXRActive.Value;
-#else
-                return true;
-#endif // UNITY_2019_3_OR_NEWER
+                return isXRSDKSuppressingLegacyXR.HasValue && isXRSDKSuppressingLegacyXR.Value;
             }
         }
 
-#if UNITY_2019_3_OR_NEWER
         /// <summary>
         /// Called when packages are installed or uninstalled, to toggle a new check on XR SDK package installation status.
         /// </summary>
-        private static void EditorApplication_projectChanged() => isLegacyXRActive = null;
+        private static void EditorApplication_projectChanged() => isXRSDKSuppressingLegacyXR = null;
+#if XR_MANAGEMENT_ENABLED
+        /// <summary>
+        /// Retrieves the enabled XRSDK XR loaders (plugins) for the current build target
+        /// </summary>
+        private static IReadOnlyList<XRLoader> XRSDKLoadersOfCurrentBuildTarget
+        {
+            get
+            {
+                BuildTargetGroup currentBuildTarget = BuildPipeline.GetBuildTargetGroup(EditorUserBuildSettings.activeBuildTarget);
+                XRGeneralSettings settingsOfCurrentTarget = XRGeneralSettingsPerBuildTarget.XRGeneralSettingsForBuildTarget(currentBuildTarget);
+#pragma warning disable CS0618 // Suppressing the warning to support xr management plugin 3.x and 4.x
```

Comment:
Is there a different way we should be getting the loaders wherever the warning is happening?

Reply:
This is intended as I do not want to add more version defines... Again this is needed to support 3.x and 4.x XR Plugin Management package.